### ISSUE TITLE
Base option in loader's config

### ISSUE THREAD

**Issue by coldshine** [zimmo-be/twig-loader#31]
It would be nice to have `base` option in loader's config in order to be able to set twig templates base path. This request is similar to [this](https://github.com/twigjs/twig.js/issues/73) in twig.js.

**Comment by webberig** [comment]
Resolving (aka finding) other twig templates is done with webpack itself, loaders shouldn't be responsible for that. Please take a look at https://github.com/webpack/docs/wiki/configuration#resolveroot

**Comment by coldshine** [comment]
Thank you for your response. 
I've checked your suggestion and it's not working with, for instance, twig `include` statements.

Consider the following structure:
```
/templates/
    a.twig
    b.twig
```

If I put `{% import './templates/b.twig %}` in `a.twig` it will search `b.twig` in `/templates/templates/b.twig` relative to `a.twig` no matter how root is resolved.

Check this line: https://github.com/zimmo-be/twig-loader/blob/master/lib/compiler.js#L12
`resourcePath` is the path of the current file. 
In my example `resourcePath = 'templates'` and `token.value = './templates/b.twig'` so `path.resolve(path.dirname(resourcePath), token.value) = '/templates/templates/b.twig'`

**Comment by webberig** [comment]
I doubt your example would work even in plain Javascript. Why not just put `./b.twig` ?

Even if there's a bug with generating the hash, I think adding a `base` option to the loader would be a bad solution to the problem...

**Comment by coldshine** [comment]
> Why not just put ./b.twig ?

You see, I'm using twig to render templates both on frontend and backend sides (with php). 
On the backend I've set my template path in `Twig_Loader_Filesystem`. 

If I'll use `./b.twig` in my `a.twig` - backend script will be using a path relative to the base path, which is set in  `Twig_Loader_Filesystem` , i.e. `/b.twig`, while frontend script will be using path relative to `a.twig`, i.e `/templates/b.twig` 

> I doubt your example would work even in plain Javascript.

I've managed to make it work by modifying `compiler.js` and `loader.js`. In case you're interested I can submit a PR.

> Even if there's a bug with generating the hash, I think adding a base option to the loader would be a bad solution to the problem...

I agree, maybe it's not a best solution. Not to mention I had to overload `Twig.compiler.compile` method in order to pass `base` option to `compiler.js`. On the other hand I couldn't find any other solution to my problem. And twig.js also has [base](https://github.com/twigjs/twig.js/issues/73#issuecomment-17522394) option for this scenario.
